beetsplug/rewrite.py

```diff
@@ -34,9 +34,7 @@ def rewriter(field, rules):
     def fieldfunc(item):
         value = item._values_fixed[field]
         for pattern, replacement in rules:
-            if pattern.match(value.lower()):
-                # Rewrite activated.
-                return replacement
+            return re.sub(pattern, replacement, value)
```

Comment:
Yeah, it is a little odd. The reason is that the original purpose of the plugin was for those "canonicalization" use cases, rather than for ordinary find-and-replace.

Since the new behavior is obviously better but breaking existing configurations could be a huge problem for some people, I lean toward the config flag plan. We could ask people to transition over to the new style and then change the default.